### ISSUE TITLE
[Firefox] doesn't work on browser start / incompatibility with StartupMaster

### ISSUE THREAD

**Issue by codydunne** [uBlock-LLC/uBlock#553]
I just installed uBlock 0.8.5.7 on Firefox 31.4.0 ESR. It worked initially, but as soon as I restarted Firefox it stopped working. There is no toolbar button and ads are not blocked.

uBlock still shows up in the Add-ons Manager and the options can be changed, except that the 3rd-party filters page fails to load and I get the Windows spinning circle loading indicator. The other options tabs are still responsive though.

**Comment by gorhill** [comment]
There is a timeout of 15s when a remote server is not responding.

The symptom you describes is as if connection to Github is blocked -- uBlock is trying to fetch the checksums file to find out what needs to be updated. So the thing to look for is whether connections to Github are blocked.

If so, waiting 15-30s the timeout should expire and the dashboard should display fine using the cached version of the checksums.

**Comment by codydunne** [comment]
I tried letting it wait for 20 minutes or so, but it still didn't display. Is the timeout failing to expire for some reason? Or is it downloading but doing it super-super slow?

**Comment by gorhill** [comment]
Ok 20 minutes is definitely not right, as said, at most 15-30 seconds.

It does appear the code is aborting when it tries to read the remote resources, thus the time-out is never fired. This would need to be reproduced and investigated using the debugger. I don't have FF 31 ESR. I will try later today installing it in a VM and see if I can reproduce.

**Comment by Deathamns** [comment]
@codydunne If the toolbar button doesn't appear, that could indicate that there is a JavaScript error which causes the initialization to fail.
Do this: Go to the Add-ons Manager (about:addons), disable µBlock, open the Browser Console (Ctrl+Shift+J), make sure that everything is check in the JS dropdown, clear the console (last button top), and now enable the extension. See what the console says.

**Comment by codydunne** [comment]
@Deathamns It looks like disabling and enabling it fixes the issue, but must be done after every restart. Here is what I get in the browser console when I enable it:

```
value is not a non-null object browser.js:14757
docShell.QueryInterface(...).sessionHistory is null content-sessionStore.js:230
value is not a non-null object browser.js:14757
1421769381517   Services.Metrics.Provider.org.mozilla.addons    WARN    Add-on type without field: locale
```

**Comment by Deathamns** [comment]
Those console reports are not interesting.
If it doesn't work when starting the browser, then simply open the console and see what it has after the browser started.

**Comment by codydunne** [comment]
It turns out that using the [StartupMaster](https://addons.mozilla.org/en-US/firefox/addon/startupmaster/) extension at the same time as uBlock is what is causing the problem. When I disable StartupMaster everything works fine.

Thank you both for your help, and I'm sorry the solution is so mundane!

**Comment by Deathamns** [comment]
That doesn't make the problem solved.
This is what I get in the console:
`NS_ERROR_FAILURE: Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIAppShellService.hiddenDOMWindow] bootstrap.js:55`

Meaning, the background page cannot be created, so the extension doesn't start at all, when StartupMaster is used.

**Comment by Deathamns** [comment]
Should be fixed in 631da22.
